p2p/node.go

```diff
@@ -107,6 +108,22 @@ func New(config Config) (*Node, error) {
 		return nil, errors.New("config.RendezvousString is required")
 	}
 
+	// HACK(albrow): As a workaround for AutoNAT issues, ping ifconfig.me to
+	// determine our public IP address on boot. This will work for nodes that
+	// would be reachable via a public IP address but don't know what it is (e.g.
+	// because they are running in a Docker container).
+	publicIP, err := getPublicIP()
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("could not get public IP address: %s", err.Error())
```

Comment:
If a node doesn't have a public IP address, shouldn't it still boot up and try to use AutoRelay?

Reply:
ifconfig.me will always return an IP address. It just might not necessarily be an IP address that you are actually publicly reachable at.

Reply:
Gotcha.